### PR TITLE
Replace Tomcat with Webflux

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR replaces the embedded Tomcat server used as the Spring Boot management server with [WebFlux](https://docs.spring.io/spring-framework/reference/web/webflux.html). While we don't particularly care whether the management server is reactive or not, the idea is to solve the issue around [JSON logging](#12886) and reduce dependencies (since we already depend on Netty and reactive-streams). Additionally, WebFlux is maintained by Spring, so we can expect it to remain better aligned with Spring Boot (e.g. logging wise, metrics, etc.), versus Tomcat.

The notable part here is that "forward:/*" routes are not supported by WebFlux. You can use HTTP redirects, but this may break some deployments which rely on our "/metrics", "/health", "/ready", etc., URLs. So instead we use a [WebFilter](https://docs.spring.io/spring-framework/reference/web/webflux/reactive-spring.html#webflux-filters) to match and forward URLs internally. That's the best I could find to simulate the old process. I do think redirects would be cleaner, but again, backwards compatibility and all.

## Related issues

related to #12886 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
There's a conflict, but I'll fix it in a bit. Anyhow, shouldn't block reviewing.

The main thing here is we cannot use the "forward:/actuator/prometheus" kind of routes, but instead must add a manual `WebFilter` bean which will do just that. What WebFlux supports is redirection, but that uses an actual HTTP redirect, which is a 3xx code, and I wasn't sure if all systems would support that kind of stuff (e.g. the k8s probe codes and so on).

**Comment by npepinpe** [comment]
I think a good follow up here would be https://github.com/camunda/zeebe/issues/8845, since, for example, the gateway now only has the probes, which are a little useless there.

**Review by lenaschoenburg — COMMENTED** [review]
:question: Do you know if forwarding has security implications? I think actuators can be put behind authentication/authorization and it would be bad if our custom "redirect" could break that.

**Comment by npepinpe** [comment]
That's a good question; I suspect it would have affected us before as well :sweat_smile:

I assume it would be safe, as our filter intercepts and reroute the request internally - it then goes through the chain. But it's good to test.

**Comment by npepinpe** [comment]
So there's more than one thing which breaks with these redirects - but note they also break with the current Tomcat ones :upside_down_face: 

Changing the base path from `/actuator` to something else will break the redirect. This also affects the current redirects with Tomcat :smile: That's fixed by resolving the `management.endpoints.web.base-path` properties and using that as the base path. However, you can also customize the path mapping for any actuator endpoint via `management.endpoints.web.path-mapping`, so you then have to resolve the path yourself for all possible redirects. Again, this also affects the existing Tomcat solution :smile: 

So, these redirects are all already easily broken in the current solution. I don't know if we want to scope here to fix them, or just avoid breaking the previous behavior.

**Comment by npepinpe** [comment]
Regarding security (TLS), since we use a single server for all routes, it will be applied to everything.

Authentication/authorization is a bit muddier here. We don't actually include Spring Security with our application, so I expect only power users would actually take the time to build their own extension JAR with Spring Security, their security chain filters, and inject that into the class path. Possible, and could also break the redirects, but at this point the fallback is simply to use the original routes - if a user is already doing such a deployment, I don't think it's too crazy to ask this of them.

All in all, I would personally vote to drop the redirects, but that's definitely a breaking change. We could keep them for 8.3, deprecate them, and remove them in 8.4 (once SaaS controller and Helm chart are updated). wdyt?

**Comment by oleschoenburg** [comment]
> I would personally vote to drop the redirects, but that's definitely a breaking change.

Do you have an overview of what this would break? If it's just SaaS and Helm I'd say it's fine to do this in 8.3, without a deprecation period. Custom baremetal deployments are rare and IMO we can expect operators to make necessary adjustments when upgrading, especially if it is as easy as changing a couple of paths.

**Comment by npepinpe** [comment]
:+1: Helm charts already use actuator routes

**Comment by npepinpe** [comment]
Controller does not use gateway health indicators (created an issue for it), but maybe uses the `/metrics` route on the broker - I'll check with them. If not, then we can remove the redirects and merge :+1:

**Comment by npepinpe** [comment]
https://github.com/camunda-cloud/camunda-operator/pull/1747 - once that's merged and deployed we can merge this.

**Comment by npepinpe** [comment]
Alright, SaaS stuff is merged, just gotta wait for it to be released :+1:

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java [hunk 1]
```diff
@@ -54,22 +53,6 @@ public static void tearDownClass() {
     sutBroker.stop();
   }
 
-  @Test
-  public void shouldGetMetrics() {
```
Comment (oleschoenburg) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java:58]: :question: Do we want to remove the entire test? Isn't it enough to just adjust the path?

**Review by lenaschoenburg — APPROVED** [review]
Cool, love to have fewer dependencies :clap:

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java (hunk 1 above)
Comment (npepinpe) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java:58]: I figured it's a built-in Spring Boot endpoint, so no need to test it. We can keep it as a sanity check if you'd like.

File: dist/src/main/java/io/camunda/zeebe/shared/management/WebFluxForwarder.java
[the remark this answers is not in the document]
Comment (npepinpe) [reply dist/src/main/java/io/camunda/zeebe/shared/management/WebFluxForwarder.java:1]: ups :see_no_evil:

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java (hunk 1 above)
Comment (oleschoenburg) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/BrokerMonitoringEndpointTest.java:58]: I think that's fair, I was just not sure why it's removed.

**Pull request merged**